### PR TITLE
Add misterdeck 3d-printed handwired stream deck with potentiometers as joystick axes

### PR DESCRIPTION
**Pull request by broekema**
This adds a QMK implementation to run on the excellent 3d printable streamdeck available here: https://www.thingiverse.com/thing:4627779. Features include 4 potentiometers that are defined as 4 joystick axes and 12 buttons. The default keymap defines the top 8 buttons as keys and the bottom 4 as joystick buttons. The 'nobuttons' keymap only defines keys and no joystick buttons (this will trigger a bug in the current version of pygame, discussed here: https://github.com/pygame/pygame/issues/2932)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by broekema** [comment]
I just noticed I need to add some more documentation (info.json, README.md). Will add to this PR shortly
.

File: keyboards/handwired/misterdeck/misterdeck.c [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "misterdeck.h"
+
+void matrix_init_kb(void) {
+  matrix_init_user();
+}
```
Comment (drashna) [inline keyboards/handwired/misterdeck/misterdeck.c:21]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/misterdeck/misterdeck.c (hunk 1 above)
Comment (broekema) [reply keyboards/handwired/misterdeck/misterdeck.c]: done in commit https://github.com/qmk/qmk_firmware/pull/15556/commits/cc306d29f6926acd96ada0014321efabb7eeb7be

File: keyboards/handwired/misterdeck/rules.mk [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
```
Comment (fauxpark) [inline keyboards/handwired/misterdeck/rules.mk:18]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/handwired/misterdeck/readme.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# MisterDeck
+
+![MisteRdeck](https://i.imgur.com/FTPWqkX.jpeg)
```
Comment (fauxpark) [inline keyboards/handwired/misterdeck/readme.md:3]:
```suggestion
![MisteRdeck](https://i.imgur.com/FTPWqkXh.jpeg)
```

File: keyboards/handwired/misterdeck/readme.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+* Hardware Availability: [Thingiverse](https://www.thingiverse.com/thing:4627779)
+
+Make example for this keyboard (after setting up your build environment):
+
+    qmk compile -kb handwired/misterdeck -km default
+    qmk flash -kb handwired/misterdeck -km default
+
+or
+
+    make handwired/misterdeck:default:flash
```
Comment (fauxpark) [inline keyboards/handwired/misterdeck/readme.md:18]:
```suggestion
    make handwired/misterdeck:default

Flashing example for this keyboard:

    make handwired/misterdeck:default:flash
```

File: keyboards/handwired/misterdeck/keymaps/nobuttons/rules.mk [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Build Options
```
Comment (fauxpark) [inline keyboards/handwired/misterdeck/keymaps/nobuttons/rules.mk:1]: This file can be deleted.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/misterdeck/keymaps/nobuttons/rules.mk (hunk 5 above)
Comment (broekema) [reply keyboards/handwired/misterdeck/keymaps/nobuttons/rules.mk]: file deleted

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**